language/vm/src/serializer.rs

```diff
@@ -854,6 +854,7 @@ impl CommonSerializer {
         binary: &mut BinaryData,
         module_handles: &[ModuleHandle],
     ) -> Result<()> {
+        precondition!(self.table_count < 1);
```

Comment:
don't understand this can you explain what we are trying to do?
Am I supposed to change this if I change the order in which the tables are serialized?
why?

Reply:
Actually, looking at this again, I think the precondition is in the wrong place. It should only be make explicit for public functions. I'll redo this.